indico/modules/events/layout/controllers/layout.py

```diff
@@ -74,12 +75,24 @@ def _process(self):
 
 
 class RHLayoutEdit(RHLayoutBase):
-    def _process(self):
+    def _process_GET(self):
         if self.event.type_ == EventType.conference:
             return self._process_conference()
         else:
             return self._process_lecture_meeting()
 
+    def _process_POST(self):
+        old_vals = self.event.get_settings('layout')
+        self._process_GET()
+        changes = {name: [old_vals.get(name), value, 'string']
```

Comment:
you can most likely use `old_vals[name]` here

Reply:
It's necessary to use `.get()`. On a freshly created event, layout settings will be empty before making any change:

```python
>>> self.event.settings.filter(EventSetting.module == 'layout').all()
[]
```

Reply:
That's why the EventSettingsProxy (e.g. `layout_settings` here) should be used. It contains all the defaults.

Reply:
Fixed 👌